filterBox.js

```diff
@@ -28,9 +28,24 @@ function byType(filteredData) {
     return filteredData;
 }
 
-//filter function by tag (Alex)
 function byTag(filteredData) {
-    return filteredData
+    const selectedButtons = [];
+    tagButtons.forEach(button => {
+        if (button.classList.contains('filter__tagButton--selected')) {
+        selectedButtons.push(button);
+        }
+    });
+
+    if (selectedButtons.length === 0) {
+        return filteredData;
+    }
+
+    const selectedTags = selectedButtons.map(button => button.innerHTML.toLowerCase());
```

Comment:
There's nothing wrong with this line of code from a syntax point of view, or even functionality-wise. But "philosophically", with relation to "separation of concerns", there's a problem here that has to do with the approach to where _state_ is stored. "State" is a term used to describe any data (usually in in-memory variables etc) that describes the state in which the application currently exists.

The way this solution is structured, as manifested on this line but also elsewhere, the UI layer (the DOM) is used to store state. That's generally not a great idea. What happens if you want to make some changes to how these buttons are presented, that requires an extra span inside them? That would break this line of code, because the `innerHTML` property would include `<span>` and `</span>`.

Ideally, the state should be stored in as simple a format as possible, e.g. a string variable somewhere, and the UI would then be built to _represent_ that state, but not be it's source of truth.

This is really nitpicking the code. Of course this code is fine as it is. But in a bigger application, this type of lack of separation of concerns would probably become a problem.